diff -Naur openssl-1.1.0e-orig/apps/build.info openssl-1.1.0e-patched/apps/build.info
--- openssl-1.1.0e-orig/apps/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/apps/build.info	2017-02-17 09:17:00.867074700 +0200
@@ -14,7 +14,7 @@
           app_rand.c \
           {- $target{apps_aux_src} -}
   INCLUDE[openssl]=.. ../include
-  DEPEND[openssl]=../libssl
+  DEPEND[openssl]=../libssl{- $config{suffix} -}
 
   SCRIPTS=CA.pl {- $tsget_name -}
   SOURCE[CA.pl]=CA.pl.in
diff -Naur openssl-1.1.0e-orig/build.info openssl-1.1.0e-patched/build.info
--- openssl-1.1.0e-orig/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/build.info	2017-02-17 09:17:00.869074800 +0200
@@ -1,9 +1,29 @@
-LIBS=libcrypto libssl
-ORDINALS[libcrypto]=crypto
-ORDINALS[libssl]=ssl
-INCLUDE[libcrypto]=. crypto/include include
-INCLUDE[libssl]=. include
-DEPEND[libssl]=libcrypto
+{-
+	if ($config{ex_libs} =~ /-static/) {
+		$config{suffix} = "MT";
+	} else {
+		$config{suffix} = "MD";
+	}
+	
+	if ($config{build_type} eq "debug") {
+		$config{suffix} .= "d";
+	}
+
+	print "==================================================\n";
+	foreach my $name (sort keys %config) { printf "%-30s %s\n", $name, $config{$name}; }
+	print "==================================================\n";
+	foreach my $name (sort keys %target) { printf "%-30s %s\n", $name, $target{$name}; }
+	print "==================================================\n";
+	
+	""
+-}
+
+LIBS=libcrypto{- $config{suffix} -} libssl{- $config{suffix} -}
+ORDINALS[libcrypto{- $config{suffix} -}]=crypto
+ORDINALS[libssl{- $config{suffix} -}]=ssl
+INCLUDE[libcrypto{- $config{suffix} -}]=. crypto/include include
+INCLUDE[libssl{- $config{suffix} -}]=. include
+DEPEND[libssl{- $config{suffix} -}]=libcrypto{- $config{suffix} -}
 
 # Empty DEPEND "indices" means the dependencies are expected to be built
 # unconditionally before anything else.
@@ -24,8 +44,6 @@
  SHARED_NAME[libcrypto]=libcrypto-{- $config{shlib_major}."_".$config{shlib_minor} -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
  SHARED_NAME[libssl]=libssl-{- $config{shlib_major}."_".$config{shlib_minor} -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
 ELSIF[{- $config{target} =~ /^VC-/ -}]
- SHARED_NAME[libcrypto]=libcrypto-{- $config{shlib_major}."_".$config{shlib_minor} -}{- $target{multilib} -}
- SHARED_NAME[libssl]=libssl-{- $config{shlib_major}."_".$config{shlib_minor} -}{- $target{multilib} -}
 ENDIF
 
 # VMS has a cultural standard where all libraries are prefixed.
@@ -39,3 +57,8 @@
  SHARED_NAME[libcrypto]=ossl$libcrypto{- sprintf "%02d%02d", $config{shlib_major}, $config{shlib_minor} -}_shr{- $target{pointer_size} -}
  SHARED_NAME[libssl]=ossl$libssl{- sprintf "%02d%02d", $config{shlib_major}, $config{shlib_minor} -}_shr{- $target{pointer_size} -}
 ENDIF
+
+
+
+
+
diff -Naur openssl-1.1.0e-orig/Configurations/00-base-templates.conf openssl-1.1.0e-patched/Configurations/00-base-templates.conf
--- openssl-1.1.0e-orig/Configurations/00-base-templates.conf	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/Configurations/00-base-templates.conf	2017-02-17 09:17:00.871074900 +0200
@@ -27,7 +27,7 @@
 	padlock_asm_src	=> "",
 	chacha_asm_src	=> "chacha_enc.c",
 	poly1305_asm_src	=> "",
-
+	suffix			=> "",
 	unistd		=> "<unistd.h>",
 	shared_target	=> "",
 	shared_cflag	=> "",
diff -Naur openssl-1.1.0e-orig/Configurations/windows-makefile.tmpl openssl-1.1.0e-patched/Configurations/windows-makefile.tmpl
--- openssl-1.1.0e-orig/Configurations/windows-makefile.tmpl	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/Configurations/windows-makefile.tmpl	2017-02-17 09:17:00.873075000 +0200
@@ -12,7 +12,7 @@
  our $dsoext = $target{dso_extension} || ".dll";
 
  our $sover = $config{shlib_major}."_".$config{shlib_minor};
-
+ 
  my $win_installenv =
      $target{build_scheme}->[2] eq "VC-W32" ?
      "ProgramFiles(x86)" : "ProgramW6432";
@@ -33,7 +33,7 @@
  sub shlib {
      return () if $disabled{shared};
      my $lib = shift;
-     return $unified_info{sharednames}->{$lib} . $shlibext;
+     return $unified_info{sharednames}->{$lib}. $shlibext;
  }
 
  sub shlib_import {
diff -Naur openssl-1.1.0e-orig/crypto/aes/build.info openssl-1.1.0e-patched/crypto/aes/build.info
--- openssl-1.1.0e-orig/crypto/aes/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/aes/build.info	2017-02-17 09:17:00.875075100 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         aes_misc.c aes_ecb.c aes_cfb.c aes_ofb.c \
         aes_ige.c aes_wrap.c {- $target{aes_asm_src} -}
 
diff -Naur openssl-1.1.0e-orig/crypto/asn1/build.info openssl-1.1.0e-patched/crypto/asn1/build.info
--- openssl-1.1.0e-orig/crypto/asn1/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/asn1/build.info	2017-02-17 09:17:00.877075300 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         a_object.c a_bitstr.c a_utctm.c a_gentm.c a_time.c a_int.c a_octet.c \
         a_print.c a_type.c a_dup.c a_d2i_fp.c a_i2d_fp.c \
         a_utf8.c a_sign.c a_digest.c a_verify.c a_mbstr.c a_strex.c \
diff -Naur openssl-1.1.0e-orig/crypto/async/build.info openssl-1.1.0e-patched/crypto/async/build.info
--- openssl-1.1.0e-orig/crypto/async/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/async/build.info	2017-02-17 09:17:00.879075400 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         async.c async_wait.c async_err.c arch/async_posix.c arch/async_win.c \
         arch/async_null.c
diff -Naur openssl-1.1.0e-orig/crypto/bf/build.info openssl-1.1.0e-patched/crypto/bf/build.info
--- openssl-1.1.0e-orig/crypto/bf/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/bf/build.info	2017-02-17 09:17:00.880075400 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=bf_skey.c bf_ecb.c bf_cfb64.c bf_ofb64.c \
+SOURCE[../../libcrypto{- $config{suffix} -}]=bf_skey.c bf_ecb.c bf_cfb64.c bf_ofb64.c \
         {- $target{bf_asm_src} -}
 
 GENERATE[bf-586.s]=asm/bf-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS) $(PROCESSOR)
diff -Naur openssl-1.1.0e-orig/crypto/bio/build.info openssl-1.1.0e-patched/crypto/bio/build.info
--- openssl-1.1.0e-orig/crypto/bio/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/bio/build.info	2017-02-17 09:17:00.882075500 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         bio_lib.c bio_cb.c bio_err.c \
         bss_mem.c bss_null.c bss_fd.c \
         bss_file.c bss_sock.c bss_conn.c \
diff -Naur openssl-1.1.0e-orig/crypto/blake2/build.info openssl-1.1.0e-patched/crypto/blake2/build.info
--- openssl-1.1.0e-orig/crypto/blake2/build.info	2017-02-16 13:58:20.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/blake2/build.info	2017-02-17 09:17:00.883075600 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         blake2b.c blake2s.c m_blake2b.c m_blake2s.c
diff -Naur openssl-1.1.0e-orig/crypto/bn/build.info openssl-1.1.0e-patched/crypto/bn/build.info
--- openssl-1.1.0e-orig/crypto/bn/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/bn/build.info	2017-02-17 09:17:00.885075700 +0200
@@ -1,12 +1,11 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         bn_add.c bn_div.c bn_exp.c bn_lib.c bn_ctx.c bn_mul.c bn_mod.c \
         bn_print.c bn_rand.c bn_shift.c bn_word.c bn_blind.c \
         bn_kron.c bn_sqrt.c bn_gcd.c bn_prime.c bn_err.c bn_sqr.c \
         {- $target{bn_asm_src} -} \
         bn_recp.c bn_mont.c bn_mpi.c bn_exp2.c bn_gf2m.c bn_nist.c \
         bn_depr.c bn_const.c bn_x931p.c bn_intern.c bn_dh.c bn_srp.c
-INCLUDE[../../libcrypto]=../../crypto/include
+INCLUDE[../../libcrypto{- $config{suffix} -}]=../../crypto/include
 
 INCLUDE[bn_exp.o]=..
 
diff -Naur openssl-1.1.0e-orig/crypto/buffer/build.info openssl-1.1.0e-patched/crypto/buffer/build.info
--- openssl-1.1.0e-orig/crypto/buffer/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/buffer/build.info	2017-02-17 09:17:00.887075800 +0200
@@ -1,2 +1 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=buffer.c buf_err.c
+SOURCE[../../libcrypto{- $config{suffix} -}]=buffer.c buf_err.c
diff -Naur openssl-1.1.0e-orig/crypto/build.info openssl-1.1.0e-patched/crypto/build.info
--- openssl-1.1.0e-orig/crypto/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/build.info	2017-02-17 09:17:00.888075900 +0200
@@ -1,6 +1,6 @@
 {- use File::Spec::Functions qw/catdir catfile/; -}
-LIBS=../libcrypto
-SOURCE[../libcrypto]=\
+
+SOURCE[../libcrypto{- $config{suffix} -}]=\
         cryptlib.c mem.c mem_dbg.c cversion.c ex_data.c cpt_err.c \
         ebcdic.c uid.c o_time.c o_str.c o_dir.c o_fopen.c \
         threads_pthread.c threads_win.c threads_none.c \
@@ -33,5 +33,5 @@
 INCLUDE[armv4cpuid.o]=.
 
 IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
-  SHARED_SOURCE[../libcrypto]=dllmain.c
+  SHARED_SOURCE[../libcrypto{- $config{suffix} -}]=dllmain.c
 ENDIF
diff -Naur openssl-1.1.0e-orig/crypto/camellia/build.info openssl-1.1.0e-patched/crypto/camellia/build.info
--- openssl-1.1.0e-orig/crypto/camellia/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/camellia/build.info	2017-02-17 09:17:00.890076000 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         cmll_ecb.c cmll_ofb.c cmll_cfb.c cmll_ctr.c \
         {- $target{cmll_asm_src} -}
 
diff -Naur openssl-1.1.0e-orig/crypto/cast/build.info openssl-1.1.0e-patched/crypto/cast/build.info
--- openssl-1.1.0e-orig/crypto/cast/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/cast/build.info	2017-02-17 09:17:00.891076100 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         c_skey.c c_ecb.c {- $target{cast_asm_src} -} c_cfb64.c c_ofb64.c
 
 GENERATE[cast-586.s]=asm/cast-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS) $(PROCESSOR)
diff -Naur openssl-1.1.0e-orig/crypto/chacha/build.info openssl-1.1.0e-patched/crypto/chacha/build.info
--- openssl-1.1.0e-orig/crypto/chacha/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/chacha/build.info	2017-02-17 09:17:00.893076200 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]={- $target{chacha_asm_src} -}
+SOURCE[../../libcrypto{- $config{suffix} -}]={- $target{chacha_asm_src} -}
 
 GENERATE[chacha-x86.s]=asm/chacha-x86.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS) $(PROCESSOR)
 GENERATE[chacha-x86_64.s]=asm/chacha-x86_64.pl $(PERLASM_SCHEME)
diff -Naur openssl-1.1.0e-orig/crypto/cmac/build.info openssl-1.1.0e-patched/crypto/cmac/build.info
--- openssl-1.1.0e-orig/crypto/cmac/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/cmac/build.info	2017-02-17 09:17:00.895076300 +0200
@@ -1,2 +1 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=cmac.c cm_ameth.c cm_pmeth.c
+SOURCE[../../libcrypto{- $config{suffix} -}]=cmac.c cm_ameth.c cm_pmeth.c
diff -Naur openssl-1.1.0e-orig/crypto/cms/build.info openssl-1.1.0e-patched/crypto/cms/build.info
--- openssl-1.1.0e-orig/crypto/cms/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/cms/build.info	2017-02-17 09:17:00.897076400 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]= \
+SOURCE[../../libcrypto{- $config{suffix} -}]= \
         cms_lib.c cms_asn1.c cms_att.c cms_io.c cms_smime.c cms_err.c \
         cms_sd.c cms_dd.c cms_cd.c cms_env.c cms_enc.c cms_ess.c \
         cms_pwri.c cms_kari.c
diff -Naur openssl-1.1.0e-orig/crypto/comp/build.info openssl-1.1.0e-patched/crypto/comp/build.info
--- openssl-1.1.0e-orig/crypto/comp/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/comp/build.info	2017-02-17 09:17:00.899076500 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]= \
+SOURCE[../../libcrypto{- $config{suffix} -}]= \
         comp_lib.c comp_err.c \
         c_zlib.c
diff -Naur openssl-1.1.0e-orig/crypto/conf/build.info openssl-1.1.0e-patched/crypto/conf/build.info
--- openssl-1.1.0e-orig/crypto/conf/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/conf/build.info	2017-02-17 09:17:00.900076600 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]= \
+SOURCE[../../libcrypto{- $config{suffix} -}]= \
         conf_err.c conf_lib.c conf_api.c conf_def.c conf_mod.c \
         conf_mall.c conf_sap.c
diff -Naur openssl-1.1.0e-orig/crypto/ct/build.info openssl-1.1.0e-patched/crypto/ct/build.info
--- openssl-1.1.0e-orig/crypto/ct/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/ct/build.info	2017-02-17 09:17:00.902076700 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]= ct_b64.c ct_err.c ct_log.c ct_oct.c ct_policy.c \
+SOURCE[../../libcrypto{- $config{suffix} -}]= ct_b64.c ct_err.c ct_log.c ct_oct.c ct_policy.c \
                          ct_prn.c ct_sct.c ct_sct_ctx.c ct_vfy.c ct_x509v3.c
diff -Naur openssl-1.1.0e-orig/crypto/des/build.info openssl-1.1.0e-patched/crypto/des/build.info
--- openssl-1.1.0e-orig/crypto/des/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/des/build.info	2017-02-17 09:17:00.904076800 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         set_key.c  ecb_enc.c  cbc_enc.c \
         ecb3_enc.c cfb64enc.c cfb64ede.c cfb_enc.c \
         ofb64ede.c ofb64enc.c ofb_enc.c \
diff -Naur openssl-1.1.0e-orig/crypto/dh/build.info openssl-1.1.0e-patched/crypto/dh/build.info
--- openssl-1.1.0e-orig/crypto/dh/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/dh/build.info	2017-02-17 09:17:00.905076900 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         dh_asn1.c dh_gen.c dh_key.c dh_lib.c dh_check.c dh_err.c dh_depr.c \
         dh_ameth.c dh_pmeth.c dh_prn.c dh_rfc5114.c dh_kdf.c dh_meth.c
diff -Naur openssl-1.1.0e-orig/crypto/dsa/build.info openssl-1.1.0e-patched/crypto/dsa/build.info
--- openssl-1.1.0e-orig/crypto/dsa/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/dsa/build.info	2017-02-17 09:17:00.907077000 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         dsa_gen.c dsa_key.c dsa_lib.c dsa_asn1.c dsa_vrf.c dsa_sign.c \
         dsa_err.c dsa_ossl.c dsa_depr.c dsa_ameth.c dsa_pmeth.c dsa_prn.c \
         dsa_meth.c
diff -Naur openssl-1.1.0e-orig/crypto/dso/build.info openssl-1.1.0e-patched/crypto/dso/build.info
--- openssl-1.1.0e-orig/crypto/dso/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/dso/build.info	2017-02-17 09:17:00.909077100 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         dso_dl.c dso_dlfcn.c dso_err.c dso_lib.c \
         dso_openssl.c dso_win32.c dso_vms.c
diff -Naur openssl-1.1.0e-orig/crypto/ec/build.info openssl-1.1.0e-patched/crypto/ec/build.info
--- openssl-1.1.0e-orig/crypto/ec/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/ec/build.info	2017-02-17 09:17:00.911077200 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         ec_lib.c ecp_smpl.c ecp_mont.c ecp_nist.c ec_cvt.c ec_mult.c \
         ec_err.c ec_curve.c ec_check.c ec_print.c ec_asn1.c ec_key.c \
         ec2_smpl.c ec2_mult.c ec_ameth.c ec_pmeth.c eck_prn.c \
diff -Naur openssl-1.1.0e-orig/crypto/engine/build.info openssl-1.1.0e-patched/crypto/engine/build.info
--- openssl-1.1.0e-orig/crypto/engine/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/engine/build.info	2017-02-17 09:17:00.912077300 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         eng_err.c eng_lib.c eng_list.c eng_init.c eng_ctrl.c \
         eng_table.c eng_pkey.c eng_fat.c eng_all.c \
         tb_rsa.c tb_dsa.c tb_dh.c tb_rand.c \
diff -Naur openssl-1.1.0e-orig/crypto/err/build.info openssl-1.1.0e-patched/crypto/err/build.info
--- openssl-1.1.0e-orig/crypto/err/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/err/build.info	2017-02-17 09:17:00.914077400 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         err.c err_all.c err_prn.c
diff -Naur openssl-1.1.0e-orig/crypto/evp/build.info openssl-1.1.0e-patched/crypto/evp/build.info
--- openssl-1.1.0e-orig/crypto/evp/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/evp/build.info	2017-02-17 09:17:00.916077500 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         encode.c digest.c evp_enc.c evp_key.c evp_cnf.c \
         e_des.c e_bf.c e_idea.c e_des3.c e_camellia.c\
         e_rc4.c e_aes.c names.c e_seed.c \
diff -Naur openssl-1.1.0e-orig/crypto/hmac/build.info openssl-1.1.0e-patched/crypto/hmac/build.info
--- openssl-1.1.0e-orig/crypto/hmac/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/hmac/build.info	2017-02-17 09:17:00.918077600 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         hmac.c hm_ameth.c hm_pmeth.c
diff -Naur openssl-1.1.0e-orig/crypto/idea/build.info openssl-1.1.0e-patched/crypto/idea/build.info
--- openssl-1.1.0e-orig/crypto/idea/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/idea/build.info	2017-02-17 09:17:00.919077700 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         i_cbc.c i_cfb64.c i_ofb64.c i_ecb.c i_skey.c
diff -Naur openssl-1.1.0e-orig/crypto/kdf/build.info openssl-1.1.0e-patched/crypto/kdf/build.info
--- openssl-1.1.0e-orig/crypto/kdf/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/kdf/build.info	2017-02-17 09:17:00.921077800 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         tls1_prf.c kdf_err.c hkdf.c
diff -Naur openssl-1.1.0e-orig/crypto/lhash/build.info openssl-1.1.0e-patched/crypto/lhash/build.info
--- openssl-1.1.0e-orig/crypto/lhash/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/lhash/build.info	2017-02-17 09:17:00.923077900 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         lhash.c lh_stats.c
diff -Naur openssl-1.1.0e-orig/crypto/md2/build.info openssl-1.1.0e-patched/crypto/md2/build.info
--- openssl-1.1.0e-orig/crypto/md2/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/md2/build.info	2017-02-17 09:17:00.925078000 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         md2_dgst.c md2_one.c
diff -Naur openssl-1.1.0e-orig/crypto/md4/build.info openssl-1.1.0e-patched/crypto/md4/build.info
--- openssl-1.1.0e-orig/crypto/md4/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/md4/build.info	2017-02-17 09:17:00.926078100 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         md4_dgst.c md4_one.c
diff -Naur openssl-1.1.0e-orig/crypto/md5/build.info openssl-1.1.0e-patched/crypto/md5/build.info
--- openssl-1.1.0e-orig/crypto/md5/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/md5/build.info	2017-02-17 09:17:00.928078200 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         md5_dgst.c md5_one.c {- $target{md5_asm_src} -}
 
 GENERATE[md5-586.s]=asm/md5-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS)
diff -Naur openssl-1.1.0e-orig/crypto/mdc2/build.info openssl-1.1.0e-patched/crypto/mdc2/build.info
--- openssl-1.1.0e-orig/crypto/mdc2/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/mdc2/build.info	2017-02-17 09:17:00.930078300 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         mdc2dgst.c mdc2_one.c
diff -Naur openssl-1.1.0e-orig/crypto/modes/build.info openssl-1.1.0e-patched/crypto/modes/build.info
--- openssl-1.1.0e-orig/crypto/modes/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/modes/build.info	2017-02-17 09:17:00.931078300 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         cbc128.c ctr128.c cts128.c cfb128.c ofb128.c gcm128.c \
         ccm128.c xts128.c wrap128.c ocb128.c \
         {- $target{modes_asm_src} -}
diff -Naur openssl-1.1.0e-orig/crypto/objects/build.info openssl-1.1.0e-patched/crypto/objects/build.info
--- openssl-1.1.0e-orig/crypto/objects/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/objects/build.info	2017-02-17 09:17:00.933078500 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         o_names.c obj_dat.c obj_lib.c obj_err.c obj_xref.c
diff -Naur openssl-1.1.0e-orig/crypto/ocsp/build.info openssl-1.1.0e-patched/crypto/ocsp/build.info
--- openssl-1.1.0e-orig/crypto/ocsp/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/ocsp/build.info	2017-02-17 09:17:00.934078500 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         ocsp_asn.c ocsp_ext.c ocsp_ht.c ocsp_lib.c ocsp_cl.c \
         ocsp_srv.c ocsp_prn.c ocsp_vfy.c ocsp_err.c v3_ocsp.c
diff -Naur openssl-1.1.0e-orig/crypto/pem/build.info openssl-1.1.0e-patched/crypto/pem/build.info
--- openssl-1.1.0e-orig/crypto/pem/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/pem/build.info	2017-02-17 09:17:00.936078600 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         pem_sign.c pem_info.c pem_lib.c pem_all.c pem_err.c \
         pem_x509.c pem_xaux.c pem_oth.c pem_pk8.c pem_pkey.c pvkfmt.c
diff -Naur openssl-1.1.0e-orig/crypto/pkcs12/build.info openssl-1.1.0e-patched/crypto/pkcs12/build.info
--- openssl-1.1.0e-orig/crypto/pkcs12/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/pkcs12/build.info	2017-02-17 09:17:00.937078700 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         p12_add.c p12_asn.c p12_attr.c p12_crpt.c p12_crt.c p12_decr.c \
         p12_init.c p12_key.c p12_kiss.c p12_mutl.c p12_sbag.c \
         p12_utl.c p12_npas.c pk12err.c p12_p8d.c p12_p8e.c
diff -Naur openssl-1.1.0e-orig/crypto/pkcs7/build.info openssl-1.1.0e-patched/crypto/pkcs7/build.info
--- openssl-1.1.0e-orig/crypto/pkcs7/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/pkcs7/build.info	2017-02-17 09:17:00.938078700 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         pk7_asn1.c pk7_lib.c pkcs7err.c pk7_doit.c pk7_smime.c pk7_attr.c \
         pk7_mime.c bio_pk7.c
diff -Naur openssl-1.1.0e-orig/crypto/poly1305/build.info openssl-1.1.0e-patched/crypto/poly1305/build.info
--- openssl-1.1.0e-orig/crypto/poly1305/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/poly1305/build.info	2017-02-17 09:17:00.940078900 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         poly1305.c {- $target{poly1305_asm_src} -}
 
 GENERATE[poly1305-sparcv9.S]=asm/poly1305-sparcv9.pl $(PERLASM_SCHEME)
diff -Naur openssl-1.1.0e-orig/crypto/rand/build.info openssl-1.1.0e-patched/crypto/rand/build.info
--- openssl-1.1.0e-orig/crypto/rand/build.info	2017-02-16 13:58:21.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/rand/build.info	2017-02-17 09:17:00.941078900 +0200
@@ -1,4 +1,3 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         md_rand.c randfile.c rand_lib.c rand_err.c rand_egd.c \
         rand_win.c rand_unix.c rand_vms.c
diff -Naur openssl-1.1.0e-orig/crypto/rc2/build.info openssl-1.1.0e-patched/crypto/rc2/build.info
--- openssl-1.1.0e-orig/crypto/rc2/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/rc2/build.info	2017-02-17 09:17:00.943079000 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         rc2_ecb.c rc2_skey.c rc2_cbc.c rc2cfb64.c rc2ofb64.c
diff -Naur openssl-1.1.0e-orig/crypto/rc4/build.info openssl-1.1.0e-patched/crypto/rc4/build.info
--- openssl-1.1.0e-orig/crypto/rc4/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/rc4/build.info	2017-02-17 09:17:00.944079100 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         {- $target{rc4_asm_src} -}
 
 GENERATE[rc4-586.s]=asm/rc4-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS) $(PROCESSOR)
diff -Naur openssl-1.1.0e-orig/crypto/rc5/build.info openssl-1.1.0e-patched/crypto/rc5/build.info
--- openssl-1.1.0e-orig/crypto/rc5/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/rc5/build.info	2017-02-17 09:17:00.945079100 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         rc5_skey.c rc5_ecb.c {- $target{rc5_asm_src} -} rc5cfb64.c rc5ofb64.c
 
 GENERATE[rc5-586.s]=asm/rc5-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS)
diff -Naur openssl-1.1.0e-orig/crypto/ripemd/build.info openssl-1.1.0e-patched/crypto/ripemd/build.info
--- openssl-1.1.0e-orig/crypto/ripemd/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/ripemd/build.info	2017-02-17 09:17:00.947079300 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         rmd_dgst.c rmd_one.c {- $target{rmd160_asm_src} -}
 
 GENERATE[rmd-586.s]=asm/rmd-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS)
diff -Naur openssl-1.1.0e-orig/crypto/rsa/build.info openssl-1.1.0e-patched/crypto/rsa/build.info
--- openssl-1.1.0e-orig/crypto/rsa/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/rsa/build.info	2017-02-17 09:17:00.948079300 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         rsa_ossl.c rsa_gen.c rsa_lib.c rsa_sign.c rsa_saos.c rsa_err.c \
         rsa_pk1.c rsa_ssl.c rsa_none.c rsa_oaep.c rsa_chk.c rsa_null.c \
         rsa_pss.c rsa_x931.c rsa_asn1.c rsa_depr.c rsa_ameth.c rsa_prn.c \
diff -Naur openssl-1.1.0e-orig/crypto/seed/build.info openssl-1.1.0e-patched/crypto/seed/build.info
--- openssl-1.1.0e-orig/crypto/seed/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/seed/build.info	2017-02-17 09:17:00.949079400 +0200
@@ -1,2 +1 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=seed.c seed_ecb.c seed_cbc.c seed_cfb.c seed_ofb.c
+SOURCE[../../libcrypto{- $config{suffix} -}]=seed.c seed_ecb.c seed_cbc.c seed_cfb.c seed_ofb.c
diff -Naur openssl-1.1.0e-orig/crypto/sha/build.info openssl-1.1.0e-patched/crypto/sha/build.info
--- openssl-1.1.0e-orig/crypto/sha/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/sha/build.info	2017-02-17 09:17:00.951079500 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         sha1dgst.c sha1_one.c sha256.c sha512.c {- $target{sha1_asm_src} -}
 
 GENERATE[sha1-586.s]=asm/sha1-586.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS) $(PROCESSOR)
diff -Naur openssl-1.1.0e-orig/crypto/srp/build.info openssl-1.1.0e-patched/crypto/srp/build.info
--- openssl-1.1.0e-orig/crypto/srp/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/srp/build.info	2017-02-17 09:17:00.952079500 +0200
@@ -1,2 +1 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=srp_lib.c srp_vfy.c
+SOURCE[../../libcrypto{- $config{suffix} -}]=srp_lib.c srp_vfy.c
diff -Naur openssl-1.1.0e-orig/crypto/stack/build.info openssl-1.1.0e-patched/crypto/stack/build.info
--- openssl-1.1.0e-orig/crypto/stack/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/stack/build.info	2017-02-17 09:17:00.953079600 +0200
@@ -1,2 +1 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=stack.c
+SOURCE[../../libcrypto{- $config{suffix} -}]=stack.c
diff -Naur openssl-1.1.0e-orig/crypto/ts/build.info openssl-1.1.0e-patched/crypto/ts/build.info
--- openssl-1.1.0e-orig/crypto/ts/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/ts/build.info	2017-02-17 09:17:00.955079700 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         ts_err.c ts_req_utils.c ts_req_print.c ts_rsp_utils.c ts_rsp_print.c \
         ts_rsp_sign.c ts_rsp_verify.c ts_verify_ctx.c ts_lib.c ts_conf.c \
         ts_asn1.c
diff -Naur openssl-1.1.0e-orig/crypto/txt_db/build.info openssl-1.1.0e-patched/crypto/txt_db/build.info
--- openssl-1.1.0e-orig/crypto/txt_db/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/txt_db/build.info	2017-02-17 09:17:00.956079800 +0200
@@ -1,2 +1 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=txt_db.c
+SOURCE[../../libcrypto{- $config{suffix} -}]=txt_db.c
diff -Naur openssl-1.1.0e-orig/crypto/ui/build.info openssl-1.1.0e-patched/crypto/ui/build.info
--- openssl-1.1.0e-orig/crypto/ui/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/ui/build.info	2017-02-17 09:17:00.957079800 +0200
@@ -1,3 +1,2 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         ui_err.c ui_lib.c ui_openssl.c ui_util.c
diff -Naur openssl-1.1.0e-orig/crypto/whrlpool/build.info openssl-1.1.0e-patched/crypto/whrlpool/build.info
--- openssl-1.1.0e-orig/crypto/whrlpool/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/whrlpool/build.info	2017-02-17 09:17:00.958079900 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=wp_dgst.c {- $target{wp_asm_src} -}
+SOURCE[../../libcrypto{- $config{suffix} -}]=wp_dgst.c {- $target{wp_asm_src} -}
 
 GENERATE[wp-mmx.s]=asm/wp-mmx.pl $(PERLASM_SCHEME) $(CFLAGS) $(LIB_CFLAGS) $(PROCESSOR)
 DEPEND[wp-mmx.s]=../perlasm/x86asm.pl
diff -Naur openssl-1.1.0e-orig/crypto/x509/build.info openssl-1.1.0e-patched/crypto/x509/build.info
--- openssl-1.1.0e-orig/crypto/x509/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/x509/build.info	2017-02-17 09:17:00.960080000 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
         x509_def.c x509_d2.c x509_r2x.c x509_cmp.c \
         x509_obj.c x509_req.c x509spki.c x509_vfy.c \
         x509_set.c x509cset.c x509rset.c x509_err.c \
diff -Naur openssl-1.1.0e-orig/crypto/x509v3/build.info openssl-1.1.0e-patched/crypto/x509v3/build.info
--- openssl-1.1.0e-orig/crypto/x509v3/build.info	2017-02-16 13:58:22.000000000 +0200
+++ openssl-1.1.0e-patched/crypto/x509v3/build.info	2017-02-17 09:17:00.961080100 +0200
@@ -1,5 +1,4 @@
-LIBS=../../libcrypto
-SOURCE[../../libcrypto]=\
+SOURCE[../../libcrypto{- $config{suffix} -}]=\
   v3_bcons.c v3_bitst.c v3_conf.c v3_extku.c v3_ia5.c v3_lib.c \
   v3_prn.c v3_utl.c v3err.c v3_genn.c v3_alt.c v3_skey.c v3_akey.c v3_pku.c \
   v3_int.c v3_enum.c v3_sxnet.c v3_cpols.c v3_crld.c v3_purp.c v3_info.c \
diff -Naur openssl-1.1.0e-orig/engines/afalg/build.info openssl-1.1.0e-patched/engines/afalg/build.info
--- openssl-1.1.0e-orig/engines/afalg/build.info	2017-02-16 13:58:23.000000000 +0200
+++ openssl-1.1.0e-patched/engines/afalg/build.info	2017-02-17 09:17:00.962080100 +0200
@@ -1,12 +1,12 @@
 IF[{- !$disabled{"engine"} -}]
   IF[{- !$disabled{afalg} -}]
     IF[{- $disabled{"dynamic-engine"} -}]
-      LIBS=../../libcrypto
-      SOURCE[../../libcrypto]=e_afalg.c e_afalg_err.c
+      LIBS=../../libcrypto{- $config{suffix} -}
+      SOURCE[../../libcrypto{- $config{suffix} -}]=e_afalg.c e_afalg_err.c
     ELSE
       ENGINES=afalg
       SOURCE[afalg]=e_afalg.c e_afalg_err.c
-      DEPEND[afalg]=../../libcrypto
+      DEPEND[afalg]=../../libcrypto{- $config{suffix} -}
       INCLUDE[afalg]= ../../include
     ENDIF
   ENDIF
diff -Naur openssl-1.1.0e-orig/engines/build.info openssl-1.1.0e-patched/engines/build.info
--- openssl-1.1.0e-orig/engines/build.info	2017-02-16 13:58:23.000000000 +0200
+++ openssl-1.1.0e-patched/engines/build.info	2017-02-17 09:17:00.963080200 +0200
@@ -1,29 +1,28 @@
 IF[{- !$disabled{"engine"} -}]
   IF[{- $disabled{"dynamic-engine"} -}]
-    LIBS=../libcrypto
-    SOURCE[../libcrypto]=\
+    SOURCE[../libcrypto{- $config{suffix} -}]=\
             e_padlock.c {- $target{padlock_asm_src} -}
     IF[{- !$disabled{capieng} -}]
-      SOURCE[../libcrypto]=e_capi.c
+      SOURCE[../libcrypto{- $config{suffix} -}]=e_capi.c
     ENDIF
   ELSE
     ENGINES=padlock
     SOURCE[padlock]=e_padlock.c {- $target{padlock_asm_src} -}
-    DEPEND[padlock]=../libcrypto
+    DEPEND[padlock]=../libcrypto{- $config{suffix} -}
     INCLUDE[padlock]=../include
     IF[{- !$disabled{capieng} -}]
       ENGINES=capi
       SOURCE[capi]=e_capi.c
-      DEPEND[capi]=../libcrypto
+      DEPEND[capi]=../libcrypto{- $config{suffix} -}
       INCLUDE[capi]=../include
     ENDIF
 
     ENGINES_NO_INST=ossltest dasync
     SOURCE[dasync]=e_dasync.c
-    DEPEND[dasync]=../libcrypto
+    DEPEND[dasync]=../libcrypto{- $config{suffix} -}
     INCLUDE[dasync]=../include
     SOURCE[ossltest]=e_ossltest.c
-    DEPEND[ossltest]=../libcrypto
+    DEPEND[ossltest]=../libcrypto{- $config{suffix} -}
     INCLUDE[ossltest]=../include
   ENDIF
 
diff -Naur openssl-1.1.0e-orig/fuzz/build.info openssl-1.1.0e-patched/fuzz/build.info
--- openssl-1.1.0e-orig/fuzz/build.info	2017-02-16 13:58:23.000000000 +0200
+++ openssl-1.1.0e-patched/fuzz/build.info	2017-02-17 09:17:00.965080300 +0200
@@ -8,6 +8,8 @@
    ""
 -}
 
+
+
 IF[{- !$disabled{"fuzz-afl"} || !$disabled{"fuzz-libfuzzer"} -}]
   PROGRAMS_NO_INST=asn1 asn1parse bignum bndiv conf crl server x509
 
@@ -21,43 +23,43 @@
 
   SOURCE[asn1]=asn1.c driver.c
   INCLUDE[asn1]=../include {- $ex_inc -}
-  DEPEND[asn1]=../libcrypto {- $ex_lib -}
+  DEPEND[asn1]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[asn1parse]=asn1parse.c driver.c
   INCLUDE[asn1parse]=../include {- $ex_inc -}
-  DEPEND[asn1parse]=../libcrypto {- $ex_lib -}
+  DEPEND[asn1parse]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[bignum]=bignum.c driver.c
   INCLUDE[bignum]=../include {- $ex_inc -}
-  DEPEND[bignum]=../libcrypto {- $ex_lib -}
+  DEPEND[bignum]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[bndiv]=bndiv.c driver.c
   INCLUDE[bndiv]=../include {- $ex_inc -}
-  DEPEND[bndiv]=../libcrypto {- $ex_lib -}
+  DEPEND[bndiv]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[cms]=cms.c driver.c
   INCLUDE[cms]=../include {- $ex_inc -}
-  DEPEND[cms]=../libcrypto {- $ex_lib -}
+  DEPEND[cms]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[conf]=conf.c driver.c
   INCLUDE[conf]=../include {- $ex_inc -}
-  DEPEND[conf]=../libcrypto {- $ex_lib -}
+  DEPEND[conf]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[crl]=crl.c driver.c
   INCLUDE[crl]=../include {- $ex_inc -}
-  DEPEND[crl]=../libcrypto {- $ex_lib -}
+  DEPEND[crl]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[ct]=ct.c driver.c
   INCLUDE[ct]=../include {- $ex_inc -}
-  DEPEND[ct]=../libcrypto {- $ex_lib -}
+  DEPEND[ct]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[server]=server.c driver.c
   INCLUDE[server]=../include {- $ex_inc -}
-  DEPEND[server]=../libcrypto ../libssl {- $ex_lib -}
+  DEPEND[server]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -} {- $ex_lib -}
 
   SOURCE[x509]=x509.c driver.c
   INCLUDE[x509]=../include {- $ex_inc -}
-  DEPEND[x509]=../libcrypto {- $ex_lib -}
+  DEPEND[x509]=../libcrypto{- $config{suffix} -} {- $ex_lib -}
 ENDIF
 
 IF[{- !$disabled{tests} -}]
@@ -73,41 +75,41 @@
 
   SOURCE[asn1-test]=asn1.c test-corpus.c
   INCLUDE[asn1-test]=../include
-  DEPEND[asn1-test]=../libcrypto
+  DEPEND[asn1-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[asn1parse-test]=asn1parse.c test-corpus.c
   INCLUDE[asn1parse-test]=../include
-  DEPEND[asn1parse-test]=../libcrypto
+  DEPEND[asn1parse-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[bignum-test]=bignum.c test-corpus.c
   INCLUDE[bignum-test]=../include
-  DEPEND[bignum-test]=../libcrypto
+  DEPEND[bignum-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[bndiv-test]=bndiv.c test-corpus.c
   INCLUDE[bndiv-test]=../include
-  DEPEND[bndiv-test]=../libcrypto
+  DEPEND[bndiv-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[cms-test]=cms.c test-corpus.c
   INCLUDE[cms-test]=../include
-  DEPEND[cms-test]=../libcrypto
+  DEPEND[cms-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[conf-test]=conf.c test-corpus.c
   INCLUDE[conf-test]=../include
-  DEPEND[conf-test]=../libcrypto
+  DEPEND[conf-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[crl-test]=crl.c test-corpus.c
   INCLUDE[crl-test]=../include
-  DEPEND[crl-test]=../libcrypto
+  DEPEND[crl-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ct-test]=ct.c test-corpus.c
   INCLUDE[ct-test]=../include
-  DEPEND[ct-test]=../libcrypto
+  DEPEND[ct-test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[server-test]=server.c test-corpus.c
   INCLUDE[server-test]=../include
-  DEPEND[server-test]=../libcrypto ../libssl
+  DEPEND[server-test]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[x509-test]=x509.c test-corpus.c
   INCLUDE[x509-test]=../include
-  DEPEND[x509-test]=../libcrypto
+  DEPEND[x509-test]=../libcrypto{- $config{suffix} -}
 ENDIF
diff -Naur openssl-1.1.0e-orig/ssl/build.info openssl-1.1.0e-patched/ssl/build.info
--- openssl-1.1.0e-orig/ssl/build.info	2017-02-16 13:58:23.000000000 +0200
+++ openssl-1.1.0e-patched/ssl/build.info	2017-02-17 09:17:00.966080300 +0200
@@ -1,5 +1,4 @@
-LIBS=../libssl
-SOURCE[../libssl]=\
+SOURCE[../libssl{- $config{suffix} -}]=\
         pqueue.c \
         statem/statem_srvr.c statem/statem_clnt.c  s3_lib.c  s3_enc.c record/rec_layer_s3.c \
         statem/statem_lib.c s3_cbc.c s3_msg.c \
diff -Naur openssl-1.1.0e-orig/test/build.info openssl-1.1.0e-patched/test/build.info
--- openssl-1.1.0e-orig/test/build.info	2017-02-16 13:58:23.000000000 +0200
+++ openssl-1.1.0e-patched/test/build.info	2017-02-17 09:17:00.969080500 +0200
@@ -20,235 +20,235 @@
 
   SOURCE[aborttest]=aborttest.c
   INCLUDE[aborttest]=../include
-  DEPEND[aborttest]=../libcrypto
+  DEPEND[aborttest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[sanitytest]=sanitytest.c
   INCLUDE[sanitytest]=../include
-  DEPEND[sanitytest]=../libcrypto
+  DEPEND[sanitytest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[bntest]=bntest.c
   INCLUDE[bntest]=.. ../crypto/include ../include
-  DEPEND[bntest]=../libcrypto
+  DEPEND[bntest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ectest]=ectest.c
   INCLUDE[ectest]=../include
-  DEPEND[ectest]=../libcrypto
+  DEPEND[ectest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ecdsatest]=ecdsatest.c
   INCLUDE[ecdsatest]=../include
-  DEPEND[ecdsatest]=../libcrypto
+  DEPEND[ecdsatest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ecdhtest]=ecdhtest.c
   INCLUDE[ecdhtest]=../include
-  DEPEND[ecdhtest]=../libcrypto
+  DEPEND[ecdhtest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[gmdifftest]=gmdifftest.c
   INCLUDE[gmdifftest]=../include
-  DEPEND[gmdifftest]=../libcrypto
+  DEPEND[gmdifftest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[pbelutest]=pbelutest.c
   INCLUDE[pbelutest]=../include
-  DEPEND[pbelutest]=../libcrypto
+  DEPEND[pbelutest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ideatest]=ideatest.c
   INCLUDE[ideatest]=../include
-  DEPEND[ideatest]=../libcrypto
+  DEPEND[ideatest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[md2test]=md2test.c
   INCLUDE[md2test]=../include
-  DEPEND[md2test]=../libcrypto
+  DEPEND[md2test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[md4test]=md4test.c
   INCLUDE[md4test]=../include
-  DEPEND[md4test]=../libcrypto
+  DEPEND[md4test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[md5test]=md5test.c
   INCLUDE[md5test]=../include
-  DEPEND[md5test]=../libcrypto
+  DEPEND[md5test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[hmactest]=hmactest.c
   INCLUDE[hmactest]=../include
-  DEPEND[hmactest]=../libcrypto
+  DEPEND[hmactest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[wp_test]=wp_test.c
   INCLUDE[wp_test]=../include
-  DEPEND[wp_test]=../libcrypto
+  DEPEND[wp_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[rc2test]=rc2test.c
   INCLUDE[rc2test]=../include
-  DEPEND[rc2test]=../libcrypto
+  DEPEND[rc2test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[rc4test]=rc4test.c
   INCLUDE[rc4test]=../include
-  DEPEND[rc4test]=../libcrypto
+  DEPEND[rc4test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[rc5test]=rc5test.c
   INCLUDE[rc5test]=../include
-  DEPEND[rc5test]=../libcrypto
+  DEPEND[rc5test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[destest]=destest.c
   INCLUDE[destest]=../include
-  DEPEND[destest]=../libcrypto
+  DEPEND[destest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[sha1test]=sha1test.c
   INCLUDE[sha1test]=../include
-  DEPEND[sha1test]=../libcrypto
+  DEPEND[sha1test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[sha256t]=sha256t.c
   INCLUDE[sha256t]=../include
-  DEPEND[sha256t]=../libcrypto
+  DEPEND[sha256t]=../libcrypto{- $config{suffix} -}
 
   SOURCE[sha512t]=sha512t.c
   INCLUDE[sha512t]=../include
-  DEPEND[sha512t]=../libcrypto
+  DEPEND[sha512t]=../libcrypto{- $config{suffix} -}
 
   SOURCE[mdc2test]=mdc2test.c
   INCLUDE[mdc2test]=../include
-  DEPEND[mdc2test]=../libcrypto
+  DEPEND[mdc2test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[rmdtest]=rmdtest.c
   INCLUDE[rmdtest]=../include
-  DEPEND[rmdtest]=../libcrypto
+  DEPEND[rmdtest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[randtest]=randtest.c
   INCLUDE[randtest]=../include
-  DEPEND[randtest]=../libcrypto
+  DEPEND[randtest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[dhtest]=dhtest.c
   INCLUDE[dhtest]=../include
-  DEPEND[dhtest]=../libcrypto
+  DEPEND[dhtest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[enginetest]=enginetest.c
   INCLUDE[enginetest]=../include
-  DEPEND[enginetest]=../libcrypto
+  DEPEND[enginetest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[casttest]=casttest.c
   INCLUDE[casttest]=../include
-  DEPEND[casttest]=../libcrypto
+  DEPEND[casttest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[bftest]=bftest.c
   INCLUDE[bftest]=../include
-  DEPEND[bftest]=../libcrypto
+  DEPEND[bftest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ssltest_old]=ssltest_old.c
   INCLUDE[ssltest_old]=.. ../include
-  DEPEND[ssltest_old]=../libcrypto ../libssl
+  DEPEND[ssltest_old]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[dsatest]=dsatest.c
   INCLUDE[dsatest]=../include
-  DEPEND[dsatest]=../libcrypto
+  DEPEND[dsatest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[exptest]=exptest.c
   INCLUDE[exptest]=../include
-  DEPEND[exptest]=../libcrypto
+  DEPEND[exptest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[rsa_test]=rsa_test.c
   INCLUDE[rsa_test]=.. ../include
-  DEPEND[rsa_test]=../libcrypto
+  DEPEND[rsa_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[evp_test]=evp_test.c
   INCLUDE[evp_test]=../include
-  DEPEND[evp_test]=../libcrypto
+  DEPEND[evp_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[evp_extra_test]=evp_extra_test.c
   INCLUDE[evp_extra_test]=../include
-  DEPEND[evp_extra_test]=../libcrypto
+  DEPEND[evp_extra_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[igetest]=igetest.c
   INCLUDE[igetest]=.. ../include
-  DEPEND[igetest]=../libcrypto
+  DEPEND[igetest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[v3nametest]=v3nametest.c
   INCLUDE[v3nametest]=../include
-  DEPEND[v3nametest]=../libcrypto
+  DEPEND[v3nametest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[crltest]=crltest.c testutil.c
   INCLUDE[crltest]=../include
-  DEPEND[crltest]=../libcrypto
+  DEPEND[crltest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[v3ext]=v3ext.c
   INCLUDE[v3ext]=../include
-  DEPEND[v3ext]=../libcrypto
+  DEPEND[v3ext]=../libcrypto{- $config{suffix} -}
 
   SOURCE[danetest]=danetest.c
   INCLUDE[danetest]=../include
-  DEPEND[danetest]=../libcrypto ../libssl
+  DEPEND[danetest]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[heartbeat_test]=heartbeat_test.c testutil.c
   INCLUDE[heartbeat_test]=.. ../include
-  DEPEND[heartbeat_test]=../libcrypto ../libssl
+  DEPEND[heartbeat_test]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[p5_crpt2_test]=p5_crpt2_test.c
   INCLUDE[p5_crpt2_test]=../include
-  DEPEND[p5_crpt2_test]=../libcrypto
+  DEPEND[p5_crpt2_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[constant_time_test]=constant_time_test.c
   INCLUDE[constant_time_test]=.. ../include
-  DEPEND[constant_time_test]=../libcrypto
+  DEPEND[constant_time_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[verify_extra_test]=verify_extra_test.c
   INCLUDE[verify_extra_test]=../include
-  DEPEND[verify_extra_test]=../libcrypto
+  DEPEND[verify_extra_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[clienthellotest]=clienthellotest.c
   INCLUDE[clienthellotest]=../include
-  DEPEND[clienthellotest]=../libcrypto ../libssl
+  DEPEND[clienthellotest]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[bad_dtls_test]=bad_dtls_test.c
   INCLUDE[bad_dtls_test]=../include
-  DEPEND[bad_dtls_test]=../libcrypto ../libssl
+  DEPEND[bad_dtls_test]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[packettest]=packettest.c
   INCLUDE[packettest]=../include
-  DEPEND[packettest]=../libcrypto
+  DEPEND[packettest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[asynctest]=asynctest.c
   INCLUDE[asynctest]=.. ../include
-  DEPEND[asynctest]=../libcrypto
+  DEPEND[asynctest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[secmemtest]=secmemtest.c
   INCLUDE[secmemtest]=../include
-  DEPEND[secmemtest]=../libcrypto
+  DEPEND[secmemtest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[srptest]=srptest.c
   INCLUDE[srptest]=../include
-  DEPEND[srptest]=../libcrypto
+  DEPEND[srptest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[memleaktest]=memleaktest.c
   INCLUDE[memleaktest]=../include
-  DEPEND[memleaktest]=../libcrypto
+  DEPEND[memleaktest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[dtlsv1listentest]=dtlsv1listentest.c
   INCLUDE[dtlsv1listentest]=.. ../include
-  DEPEND[dtlsv1listentest]=../libssl
+  DEPEND[dtlsv1listentest]=../libssl{- $config{suffix} -}
 
   SOURCE[ct_test]=ct_test.c testutil.c
   INCLUDE[ct_test]=../crypto/include ../include
-  DEPEND[ct_test]=../libcrypto
+  DEPEND[ct_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[threadstest]=threadstest.c
   INCLUDE[threadstest]=.. ../include
-  DEPEND[threadstest]=../libcrypto
+  DEPEND[threadstest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[afalgtest]=afalgtest.c
   INCLUDE[afalgtest]=.. ../include
-  DEPEND[afalgtest]=../libcrypto
+  DEPEND[afalgtest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[d2i_test]=d2i_test.c testutil.c
   INCLUDE[d2i_test]=.. ../include
-  DEPEND[d2i_test]=../libcrypto
+  DEPEND[d2i_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ssl_test_ctx_test]=ssl_test_ctx_test.c ssl_test_ctx.c testutil.c
   INCLUDE[ssl_test_ctx_test]=.. ../include
-  DEPEND[ssl_test_ctx_test]=../libcrypto
+  DEPEND[ssl_test_ctx_test]=../libcrypto{- $config{suffix} -}
 
   SOURCE[ssl_test]=ssl_test.c ssl_test_ctx.c testutil.c handshake_helper.c
   INCLUDE[ssl_test]=.. ../include
-  DEPEND[ssl_test]=../libcrypto ../libssl
+  DEPEND[ssl_test]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[cipherlist_test]=cipherlist_test.c testutil.c
   INCLUDE[cipherlist_test]=.. ../include
-  DEPEND[cipherlist_test]=../libcrypto ../libssl
+  DEPEND[cipherlist_test]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   INCLUDE[testutil.o]=..
   INCLUDE[ssl_test_ctx.o]=../include
@@ -257,31 +257,31 @@
 
   SOURCE[x509aux]=x509aux.c
   INCLUDE[x509aux]=../include
-  DEPEND[x509aux]=../libcrypto
+  DEPEND[x509aux]=../libcrypto{- $config{suffix} -}
 
   SOURCE[asynciotest]=asynciotest.c ssltestlib.c
   INCLUDE[asynciotest]=../include
-  DEPEND[asynciotest]=../libcrypto ../libssl
+  DEPEND[asynciotest]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[bioprinttest]=bioprinttest.c
   INCLUDE[bioprinttest]=../include
-  DEPEND[bioprinttest]=../libcrypto
+  DEPEND[bioprinttest]=../libcrypto{- $config{suffix} -}
 
   SOURCE[sslapitest]=sslapitest.c ssltestlib.c testutil.c
   INCLUDE[sslapitest]=../include ..
-  DEPEND[sslapitest]=../libcrypto ../libssl
+  DEPEND[sslapitest]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[dtlstest]=dtlstest.c ssltestlib.c testutil.c
   INCLUDE[dtlstest]=../include .
-  DEPEND[dtlstest]=../libcrypto ../libssl
+  DEPEND[dtlstest]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[sslcorrupttest]=sslcorrupttest.c ssltestlib.c testutil.c
   INCLUDE[sslcorrupttest]=../include .
-  DEPEND[sslcorrupttest]=../libcrypto ../libssl
+  DEPEND[sslcorrupttest]=../libcrypto{- $config{suffix} -} ../libssl{- $config{suffix} -}
 
   SOURCE[bio_enc_test]=bio_enc_test.c
   INCLUDE[bio_enc_test]=../include
-  DEPEND[bio_enc_test]=../libcrypto
+  DEPEND[bio_enc_test]=../libcrypto{- $config{suffix} -}
 
   IF[{- !$disabled{shared} -}]
     PROGRAMS_NO_INST=shlibloadtest
@@ -311,7 +311,7 @@
   GENERATE[buildtest_$name.c]=generate_buildtest.pl $name
   SOURCE[buildtest_$name]=buildtest_$name.c
   INCLUDE[buildtest_$name]=../include
-  DEPEND[buildtest_$name]=../libssl ../libcrypto
+  DEPEND[buildtest_$name]=../libssl$config{suffix} ../libcrypto$config{suffix}
 _____
    }
 -}
diff -Naur openssl-1.1.0e-orig/test/recipes/90-test_shlibload.t openssl-1.1.0e-patched/test/recipes/90-test_shlibload.t
--- openssl-1.1.0e-orig/test/recipes/90-test_shlibload.t	2017-02-16 13:58:24.000000000 +0200
+++ openssl-1.1.0e-patched/test/recipes/90-test_shlibload.t	2017-02-17 09:17:00.970080600 +0200
@@ -22,10 +22,14 @@
 
 plan tests => 3;
 
+my $libcrypto_index="libcrypto$config{suffix}";
+
+my $libssl_index="libssl$config{suffix}";
+
 my $libcrypto =
-    $unified_info{sharednames}->{libcrypto}.$target{shared_extension_simple};
+    $unified_info{sharednames}->{$libcrypto_index}.$target{shared_extension_simple};
 my $libssl =
-    $unified_info{sharednames}->{libssl}.$target{shared_extension_simple};
+    $unified_info{sharednames}->{$libssl_index}.$target{shared_extension_simple};
 
 ok(run(test(["shlibloadtest", "-crypto_first", $libcrypto, $libssl])),
    "running shlibloadtest -crypto_first");
diff -Naur openssl-1.1.0e-orig/util/mkdef.pl openssl-1.1.0e-patched/util/mkdef.pl
--- openssl-1.1.0e-orig/util/mkdef.pl	2017-02-16 13:58:24.000000000 +0200
+++ openssl-1.1.0e-patched/util/mkdef.pl	2017-02-17 09:17:00.972080700 +0200
@@ -170,13 +170,13 @@
 		$zlib = 1;
 	}
 
-	$do_ssl=1 if $_ eq "libssl";
-	if ($_ eq "ssl") {
+	$do_ssl=1 if $_ =~ m/^libssl/;
+	if ($_ =~ m/^ssl/) {
 		$do_ssl=1; 
 		$libname=$_
 	}
-	$do_crypto=1 if $_ eq "libcrypto";
-	if ($_ eq "crypto") {
+	$do_crypto=1 if $_ =~ m/^libcrypto/;
+	if ($_ =~ m/^crypto/) {
 		$do_crypto=1;
 		$libname=$_;
 	}
